scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt

```diff
@@ -98,16 +99,18 @@ abstract class AbstractStorageTest : StringSpec() {
         scannerEndTime1,
         1,
         sortedSetOf(
-            LicenseFindings("license 1.1", sortedSetOf(), sortedSetOf()),
-            LicenseFindings("license 1.2", sortedSetOf(), sortedSetOf())
+            LicenseFinding("license 1.1", TextLocation("", -1, -1)),
```

Comment:
Please don't create "invalid" TextLocations with "magic" numbers. Just use a non-empty dummy TextLocation instead.

Reply:
Do you mean I should create a variable e.g. EMPTY_TEXT_LOCATION and use it?

Reply:
No. For tests I believe you should come up with a fake / dummy path and start / end lines, just like we use a fake / dummy license name here. I.e. use `LicenseFinding("license 1.1", TextLocation("path/to/some/file", 42, 66))`.

Reply:
I meant to put the variable underneath the test file of course.
Can you tell how your proposal is then different from mine? ...or are they similar now ?

Reply:
My general point is: IMO we should disallow the creation of invalid TextLocations. By (at least my) definition of a text location, line numbers should be >= 1. Even if we do not enforce that technically yet, having a line number of `-1` semantically does not make sense, and it's bad practice to reserve such "magic" numbers for a special purpose, like describing "no text location". If there is no text location you simply should not provide one. Unfortunately, that's currently not possible with this new model.

The next question is whether we even have the need to describe "no text location". If all our currently supported scanners provide the text location (and we just need to implement extracting it), we're fine. If some of our supported scanners do not support reporting the text location at all, we may want to consider dropping support for them.

Or, we fine a sane way to express "no text location" other than defining `EMPTY_TEXT_LOCATION = TextLocation("", -1, -1)`, e.g. by making `location` in `LicenseFinding` nullable.

Reply:
> If some of our supported scanners do not support reporting the text location at all, we may want to consider dropping support for them.

Dropping support is something I thought about as well because of maintainance cost for what value?

Are you then fine (in scope of this PR) to have a dummy location var in the test or not ?

Reply:
> Are you then fine (in scope of this PR) to have a dummy location var in the test or not ?

"Dummy" in the sense that the path is non-empty but fake, and line numbers are >= 1, then yes.

Reply:
done.

Reply:
hm, it seems the NoticeReporterTest now runs into that issue.
So the used inputs (OrtResult) contain only findings without locations, thus it kind of needs to be addressed somehow in this PR.